tests/components/gree/test_climate.py

```diff
@@ -178,6 +176,10 @@ async def test_discovery_add_device_after_setup(
     discovery.return_value.mock_devices = [MockDevice1]
     device.side_effect = [MockDevice1]
 
+    await hass.config.async_set_time_zone("UTC")
+    freezer.move_to("2021-01-09 12:00:00+00:00")
```

Comment:
I think you can also use the freeze time decorator. Please check withings test_sensor.py and then the first test, it has a freeze time annotation, maybe that helps